lib/apiservers/engine/backends/container.go

```diff
@@ -1356,10 +1356,30 @@ func validateCreateConfig(config *types.ContainerCreateConfig) error {
 
 	// validate port bindings
 	if config.HostConfig != nil {
+		var ips []string
+		if addrs, err := clientIPv4Addrs(); err != nil {
```

Comment:
We should also allow the external address to be specified explicitly given that's the one we'd be using by default anyway.